src/main/java/com/sap/sgs/phosphor/fosstars/data/github/UsesGitHubPlatform.java

```diff
@@ -0,0 +1,65 @@
+package com.sap.sgs.phosphor.fosstars.data.github;
+
+import static com.sap.sgs.phosphor.fosstars.model.feature.oss.OssFeatures.USES_GITHUB_PLATFORM;
+
+import com.sap.sgs.phosphor.fosstars.model.Value;
+import com.sap.sgs.phosphor.fosstars.model.ValueSet;
+import com.sap.sgs.phosphor.fosstars.model.value.BooleanValue;
+import com.sap.sgs.phosphor.fosstars.tool.github.GitHubProject;
+import java.io.IOException;
+import java.util.Optional;
+import org.apache.commons.lang3.StringUtils;
+import org.kohsuke.github.GHRepository;
+import org.kohsuke.github.GitHub;
+
+/**
+ * This data provider checks if a project uses GitHub as a main development platform. The check
+ * would be based on mirror url is null or empty. This will ensure whether the underlying project is
+ * a mirror project or not.
+ */
+public class UsesGitHubPlatform extends AbstractGitHubDataProvider {
+
+  /**
+   * Initializes a data provider.
+   *
+   * @param github An interface to the GitHub API.
+   */
+  public UsesGitHubPlatform(GitHub github) {
+    super(github);
+  }
+
+  @Override
+  protected UsesGitHubPlatform doUpdate(GitHubProject project, ValueSet values) throws IOException {
+    logger.info("Figuring out if the project uses GitHub as the main development platform ...");
+
+    Optional<Value> something = cache.get(project, USES_GITHUB_PLATFORM);
+    if (something.isPresent()) {
+      values.update(something.get());
+      return this;
+    }
+
+    Value<Boolean> usesGitHubPlatform = usesGitHubPlatform(project);
+    values.update(usesGitHubPlatform);
+    cache.put(project, usesGitHubPlatform, tomorrow());
+
+    return this;
+  }
+
+  /**
+   * Checks if the GitHub repository has an empty or null mirror url.
+   *
+   * @param project The project.
+   * @return {@link BooleanValue} value of the feature.
+   */
+  private Value<Boolean> usesGitHubPlatform(GitHubProject project) {
+    // TODO: This is just a basic check, more constraints have to be considered in the future with
+    //       more clarity.
+    try {
+      GHRepository repository = github.getRepository(project.path());
+      return USES_GITHUB_PLATFORM.value(StringUtils.isEmpty(repository.getMirrorUrl()));
```

Comment:
If I understand correctly, `getMirrorUrl()` returns an URL to the main repository, doesn't it?

Do you have an example of a repository which has a non-empty `mirror_url` field? I tried the repository mentioned in https://github-api.kohsuke.org/apidocs/org/kohsuke/github/GHRepository.html#getMirrorUrl() but it doesn't have this field:

```
$ curl -i https://api.github.com/repos/apache/tomee | grep mirror
  "mirror_url": null,
```

Reply:
This can be used https://github.com/coreutils/coreutils to verify.

Reply:
Good. Could you please add a method `isMirror(GitHubProject)` which returns true (boolean) if a project is a mirror, and false otherwise? Please describe in javadoc how it works. I think it would be useful since it's not very clear what `getMirrorUrl()` means.